File: backend/app/services/setting_service.py
from sqlalchemy.orm import Session
from typing import List, Dict, Optional
from ..models.setting import Setting

class SettingService:
    @staticmethod
    def set_root_directory(db: Session, directory_path: str):
        setting = db.query(Setting).filter(Setting.key == "root_directory").first()
        if setting:
            setting.value = directory_path
        else:
            setting = Setting(key="root_directory", value=directory_path)
            db.add(setting)
        db.commit()

    @staticmethod
    def get_root_directory(db: Session) -> str:
        setting = db.query(Setting).filter(Setting.key == "root_directory").first()
        return setting.value if setting else None
    
    @staticmethod
    def set_videos_per_page(db: Session, videos_per_page: int):
        setting = db.query(Setting).filter(Setting.key == "videos_per_page").first()
        if setting:
            setting.value = str(videos_per_page)
        else:
            setting = Setting(key="videos_per_page", value=str(videos_per_page))
            db.add(setting)
        db.commit()
    
    @staticmethod
    def get_videos_per_page(db: Session) -> int:
        setting = db.query(Setting).filter(Setting.key == "videos_per_page").first()
        return int(setting.value) if setting else 20  # 默认每页20个视频
    
    @staticmethod
    def set_setting(db: Session, key: str, value: str):
        setting = db.query(Setting).filter(Setting.key == key).first()
        if setting:
            setting.value = value
        else:
            setting = Setting(key=key, value=value)
            db.add(setting)
        db.commit()
    
    @staticmethod
    def get_setting(db: Session, key: str) -> Optional[str]:
        setting = db.query(Setting).filter(Setting.key == key).first()
        return setting.value if setting else None
        
    @staticmethod
    def get_all_settings(db: Session) -> List[Dict[str, str]]:
        settings = db.query(Setting).all()
        return [{'key': setting.key, 'value': setting.value} for setting in settings]

File: frontend/src/services/setting_service.js
import apiClient from './api_client'

class SettingService {
  // 获取每页视频数量设置
  async getVideosPerPage() {
    try {
      const response = await apiClient.get('/settings/videos_per_page')
      return response.data.videos_per_page
    } catch (error) {
      console.error('获取每页视频数量设置失败:', error)
      // 如果获取失败，返回默认值
      return 20
    }
  }

  // 设置每页视频数量
  async setVideosPerPage(value) {
    const response = await apiClient.post('/settings/videos_per_page', {
      videos_per_page: value
    })
    return response.data
  }

  // 获取通用设置
  async getSetting(key) {
    const response = await apiClient.get(`/settings/${key}`)
    return response.data
  }

  // 设置通用设置
  async setSetting(key, value) {
    const response = await apiClient.post(`/settings/${key}`, {
      key,
      value
    })
    return response.data
  }

  // 获取根目录设置
  async getRootDirectory() {
    try {
      const response = await apiClient.get('/settings/root_directory')
      return response.data.root_directory
    } catch (error) {
      console.error('获取根目录设置失败:', error)
      throw error
    }
  }

  // 设置根目录
  async setRootDirectory(path) {
    const response = await apiClient.post('/settings/root_directory', {
      root_directory: path
    })
    return response.data
  }

  // 扫描视频
  async scanVideos() {
    const response = await apiClient.get('/videos/scan')
    return response.data
  }

  // 获取扫描进度
  async getScanProgress() {
    const response = await apiClient.get('/videos/scan/progress')
    return response.data
  }
}

export default new SettingService()